### PR TITLE
Addon-docs: Fix React Profiler in source snippets

### PR DESCRIPTION
**Pull request by zhyd1997**
Issue: #11554 

## What I did
- [x] enabled `code/renderers` unit tests.
- [x] React Profiler support.
- [x] added stories.

## How to test

- [x] Is this testable with Jest or Chromatic screenshots?
- [x] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

File: code/renderers/react/src/docs/jsxDecorator.tsx [hunk 1]
```diff
@@ -101,6 +101,7 @@ export const renderJsx = (code: React.ReactElement, options: JSXOptions) => {
           // To get exotic component names resolving properly
           displayName: (el: any): string =>
             el.type.displayName ||
+            (el.type === Symbol.for('react.profiler') ? 'Profiler' : null) ||
```
Comment (zhyd1997) [inline code/renderers/react/src/docs/jsxDecorator.tsx:104]: the major change here!

File: code/renderers/react/src/docs/jsxDecorator.tsx [hunk 2]
```diff
@@ -141,7 +142,7 @@ export const renderJsx = (code: React.ReactElement, options: JSXOptions) => {
     return string;
   }).join('\n');
 
-  return result.replace(/function\s+noRefCheck\(\)\s+\{\}/, '() => {}');
+  return result.replace(/function\s+noRefCheck\(\)\s+\{\}/g, '() => {}');
```
Comment (zhyd1997) [inline code/renderers/react/src/docs/jsxDecorator.tsx:145]: Before:
Screen Shot 2022-08-25 at 12 47 42

After:
Screen Shot 2022-08-25 at 12 46 58

**Review by kylegach — COMMENTED** [review]
Hey, @zhyd1997! This is an amazing first-time contribution! 🥳 

Would you mind checking to see if the change related to `noRefCheck` is tested, and adding a test, if not. Thanks!

File: code/renderers/svelte/src/docs/extractArgTypes.test.ts [hunk 3]
```diff
@@ -87,7 +87,7 @@ describe('Extracting Arguments', () => {
           "control": Object {
             "type": "boolean",
           },
-          "description": null,
```
Comment (kylegach) [inline code/renderers/svelte/src/docs/extractArgTypes.test.ts:90]: @zhyd1997 — Can you please share why this change was necessary?

**Comment by zhyd1997** [comment]
> Hey, @zhyd1997! This is an amazing first-time contribution! 🥳
> 
> Would you mind checking to see if the change related to `noRefCheck` is tested, and adding a test, if not. Thanks!

I need to add a story to test it, but don't know which example under `examples` folder should I take `cra-ts-kitchen-sink` or `react-ts`?

File: code/renderers/svelte/src/docs/extractArgTypes.test.ts (hunk 3 above)
Comment (zhyd1997) [reply code/renderers/svelte/src/docs/extractArgTypes.test.ts:90]: I didn't change svelte code, and I think they 're all falsy value, so I just ran something like `update-snapshots` command.

Comment (zhyd1997) [reply code/renderers/svelte/src/docs/extractArgTypes.test.ts:90]: @kylegach 

Previously we didn't enable `code/renderers` unit test after moved all source code into `code` folder, and I need run unit test for `react`, so I manually enabled it in this PR, but `svelte` unit test failed, that's why I updated all `code/renders` snapshots here.

**Comment by iosifnicolae2** [comment]
Any news on this issue? It would be awesome if we can gave good code export functionality.

Thank you!

**Review by shilman — APPROVED** [review]
Thank you @zhyd1997 !!! 🙌

**Comment by 365kim** [comment]
Is this fix available in any version of `@storybook/react`? 🙏

**Comment by zhyd1997** [comment]
> Is this fix available in any version of `@storybook/react`? 🙏

Sorry no, available from this version: [v7.0.0-alpha.46](https://github.com/storybookjs/storybook/releases/tag/v7.0.0-alpha.46)

File: code/renderers/react/src/docs/jsxDecorator.tsx (hunk 2 above)
Comment (ns-kutsav) [reply code/renderers/react/src/docs/jsxDecorator.tsx:123]: It still isn't showing the actual function. Is that the expectedd behavior?
